src/types/index.d.ts

```diff
@@ -180,3 +180,20 @@ export type FilterExpressionRange = {
 };
 
 export type FilterExpressionRangeValue = ['-inf' | number, 'inf' | number];
+
+export interface Item extends Record<string, any> {
+  value: string;
+  is_slotted: boolean;
+  labels: Record<string, unknown>;
+  matched_terms: string[];
+  data?: Record<string, any>;
+  section?: string;
+}
+
+export interface Product extends Item {
```

Comment:
Autocomplete UI goes into more detail about the type. Should we do something similar here or do you think this level of specificity is enough?

```js
export type Product = {
  data: {
    facets: { name: string; values: string[] }[];
    group_ids: string[];
    id: string;
    image_url: string;
    url: string;
  };
  is_slotted: boolean;
  labels: Record<string, unknown>;
  matched_terms: string[];
  value: string;
  section: 'Products';
};

export type SearchSuggestion = {
  data: {
    id: string;
    url?: string;
  };
  id: string;
  is_slotted: boolean;
  labels: Record<string, unknown>;
  matched_terms: string[];
  value: string;
  section: 'Search Suggestions';
};
```

Reply:
yes please, for example: `image_url` exists on Products by not on SearchSuggestions

Reply:
Interesting. Just so we're on the same page, by default we would have these [optional, default parameters](https://docs.constructor.io/rest_api/autocomplete_queries/#products-common-fields) for `data` and if the customer has additional metadata they'd simply extend over it? 

It would probably make sense to have a base data class too then so you could do something like:

```
// In consumer code
export interface MySpecificDataObj extends BaseData {
    stars: number
}
export interface Product extends BaseProduct {
    data: MySpecificDataObj
}
```

Reply:
Made the recommended changes but also, @mocca102 and I had a pretty interesting discussion so I'm just going to record it here: 

With this PR, you'll be able to use your own version of `Products` or `Items` when you extend these types within your codebase and these types will be compatible when passed to Client-JS' functions.

However, when receiving these types back from our server, they'll default back to the default `Products` and `Items` types defined here. One solution to this would be to modify the existing function type definitions to be generic which allows consumers of this library to modify the types Constructor will be returning. This is a pretty nifty pattern that many TypeScript libraries follow, but we both agreed that the value doesn't justify the cost of implementing such a pattern. Still pretty interesting though.

Reply:
agreed, the idea of applying this generics pattern here is very interesting. thanks for explaining this here, @Mudaafi . can imagine a world where we'll want to do this down the road, but agree with the decision to wait until that is truely needed and adding real value for real customer needs